nowcasting_dataset/data_sources/pv/pv_data_source.py

```diff
@@ -72,9 +78,24 @@ def get_data_model_for_batch():
 
     def _load_metadata(self):
 
-        logger.debug(f"Loading PV metadata from {self.metadata_filename}")
+        logger.debug(f"Loading PV metadata from {self.metadata_filenames}")
+
+        # collect all metadata together
+        pv_metadata = []
+        for i, metadata_filename in enumerate(self.metadata_filenames):
+
+            # read metadata file
+            metadata = pd.read_csv(metadata_filename, index_col="system_id")
 
-        pv_metadata = pd.read_csv(self.metadata_filename, index_col="system_id")
+            # add i to index, to make sure the indexes are unique
+            new_index = metadata.index.astype(str)
+            new_index = new_index + f"_{i}"
```

Comment:
Good thinking about ensuring the PV system IDs remain unique!

However, a shortcoming of using `<pv id>_<i>`  is that `i` depends on the (arbitrary) order of the filenames in the config YAML. So, when the code manually cleans some specific PV systems (line 152), the code can't unambiguously identify, for example, "PVOutput.org system #123".

Is there a way to guarantee that, for example, PVOutput.org systems will always have the suffix `0` and PassivSystem PV systems will always have the suffix `1`?

Or maybe use a [Pandas MultiIndex](https://pandas.pydata.org/pandas-docs/stable/user_guide/advanced.html), where one level of the multi-index is `"PVOutput.org"` or `"PassivSystems"`, and the other level is the integer PV system ID?  (But I don't know if the downstream code will like using MultiIndexes!)

Reply:
we could order the PV filenames list by alphabetic order, then it should always be one way round. We could do this in the pydantic model. Have to make sure the metdata is ordered the same way

Reply:
Hmm, I gotta admit that feels fragile. What happens if we only use PVOutput.org data (which, when using both PVOutput.org and PassivSystems data would be the second entry. But would be the first (and only) entry if we only use PVOutput.org data)

How about something like detecting if the string "pvoutput" or "passiv" is in the `filename.lower()` string, and then appending either "pvoutput" or "passiv" to every PV system ID?

Reply:
Yea perhaps the model should be something like this
pv_files: [
- pv_filename:
- pv_metedata
- label
]

Then we can use the label to do the new ids. We could even add a validator to make sure all the lables are different

Reply:
SGTM!

Reply:
It certainly can!

Reply:
I've tidied up the config model, as described above. 

I then used an encode (full code [here](https://github.com/openclimatefix/nowcasting_dataset/blob/issue/554-both-pv/nowcasting_dataset/data_sources/pv/pv_data_source.py#L471)):
```
label_index = PV_PROVIDERS.index(label)
new_index = [str(int(col) * 10 + label_index) for col in indexes]
```

I've added an assert if the list of pv provers ever is more than 10 (i don't think this will happen)